pkg/fanal/image/daemon/containerd.go

```diff
@@ -56,8 +57,24 @@ func imageWriter(client *containerd.Client, img containerd.Image) imageSave {
 			return nil, xerrors.New("no image reference")
 		}
 		imgOpts := archive.WithImage(client.ImageService(), ref[0])
-		manifestOpts := archive.WithManifest(img.Target())
-		platOpts := archive.WithPlatform(platforms.DefaultStrict())
+		target := img.Target()
+		manifestOpts := archive.WithManifest(target)
+
+		manifest, err := images.Manifest(ctx, img.ContentStore(), target, img.Platform())
+		if err != nil {
+			return nil, xerrors.Errorf("error getting image manifest: %w", err)
+		}
+
+		ps, err := images.Platforms(ctx, img.ContentStore(), manifest.Config)
+		if err != nil {
+			return nil, xerrors.Errorf("error getting image platforms: %w", err)
+		}
+
+		if len(ps) == 0 {
+			return nil, xerrors.New("no platform")
+		}
+
+		platOpts := archive.WithPlatform(platforms.OnlyStrict(ps[0]))
```

Comment:
> Scan will most likely fail if a host pulls a compatible version of a multi-arch image if this version is not preferred.

I also thought about it.

We have [Platform](https://github.com/aquasecurity/trivy/blob/72e302cf818cbb2c983c64a3929778b0b25ccb1d/pkg/fanal/types/image.go#L89) options. What if we use it to fix this case.

Reply:
Hi @DmitriyLewen,
Thanks for the comment, perhaps we could use the `Platform` option only if it is set. I need to check if it has a default value that is not an empty string but otherwise it looks doable.

Reply:
@DmitriyLewen We should now use Platform for containerd images 👍